packages/dx-react-scheduler/src/plugins/date-navigator.tsx

```diff
@@ -23,29 +25,37 @@ const navigate = (action, currentView, intervalCount) => (direction, nextDate) =
   step: currentView.type,
 });
 
-export class DateNavigator extends React.PureComponent {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      visible: false,
-    };
+class DateNavigatorBase extends React.PureComponent<DateNavigatorProps, DateNavigatorState> {
+  target!: React.ReactInstance;
 
-    this.handleVisibilityToggle = this.handleVisibilityToggle.bind(this);
-    this.handleHide = this.handleHide.bind(this);
-    this.setRootRef = this.setRootRef.bind(this);
-  }
+  state = {
+    visible: false,
+  };
+  static components: PluginComponents = {
+    rootComponent: 'Root',
+    overlayComponent: 'Overlay',
+    openButtonComponent: 'OpenButton',
+    navigationButtonComponent: 'NavigationButton',
+    calendarComponent: 'Calendar',
+    calendarRowComponent: 'CalendarRow',
+    calendarCellComponent: 'CalendarCell',
+    calendarHeaderRowComponent: 'CalendarHeaderRow',
+    calendarHeaderCellComponent: 'CalendarHeaderCell',
+    calendarTextComponent: 'CalendarText',
+    calendarNavigatorComponent: 'CalendarNavigator',
+    calendarNavigationButtonComponent: 'CalendarNavigationButton',
+  };
 
-  setRootRef(target) {
+  setRootRef = (target) => {
     this.target = target;
   }
 
-  handleVisibilityToggle() {
+  handleVisibilityToggle = () => {
     const { visible } = this.state;
     this.setState({ visible: !visible });
```

Comment:
Use callback version of `setState` as new state is calculated from the current.
https://reactjs.org/docs/react-component.html#setstate